components/raftstore/src/store/peer_storage.rs

```diff
@@ -631,7 +672,15 @@ where
             "state" => ?self.apply_state(),
         );
 
-        Ok(region)
+        Ok(HandleReadyResult::Snapshot(Box::new(
```

Comment:
Why not return `(region, for_witness)`? So we can keep `apply_snapshot` itself not tied to handle result and easier to test.